### ISSUE TITLE
Enhancement: update Makefiles to install all man pages

### ISSUE THREAD

**Issue by xexyl** [ioccc-src/mkiocccentry#453]
As I put in comment https://github.com/ioccc-src/mkiocccentry/issues/448#issuecomment-1368083963 the Makefiles do not install all man pages for `make install`. Specifically the man pages that link to another man page are not installed.

Now I have the dbg/Makefile modifications so that doing `make -C dbg install` would actually install the man pages but I did not do it all in `man/Makefile` as I ran out of time at that point. Since then the Makefiles have been changed quite a bit but I do have the changes here.

However as comment https://github.com/ioccc-src/mkiocccentry/issues/448#issuecomment-1368287638 suggests I am opening this issue to better address it.

I am being called away now so I have to keep this as is for now. I can elaborate more later.

**Comment by xexyl** [comment]
I can hopefully at least do some of this tomorrow. I added some comments about why make test is failing for you and not me and how to fix it. 

But I can do more tomorrow so that nobody else will have the problem either. The comments should explain how you might do it too. 

It is because I forgot to add a file.

**Comment by xexyl** [comment]
And with that I must go for the night. Happy new year and stay safe! More from me tomorrow. Good night!

**Comment by lcn2** [comment]
Thanks for your help on this issue, @xexyl .. with the new issue you will open, we can close this issue now.

**Comment by xexyl** [comment]
> Thanks for your help on this issue, @xexyl .. with the new issue you will open, we can close this issue now.

Though I think that this issue is not yet resolved. I know what it takes to make it so man pages will be installed but it is not yet done. 

The other man pages issue is not one I am entirely sure how to do. 

Expect more from me in a bit. The fix for the make test failing will take very little time at all but I have other things to do first and in fact I am going to start doing those things now. 

Back later!

**Comment by xexyl** [comment]
I think this was closed prematurely. This is proving to be a problem. Although it's simple for dbg/Makefile itself the man/Makefile is much more complicated here because it wants to copy the man pages from the other directories to itself. This complicates the man/Makefile a lot.

I can upload the Makefile that I have which is the changes so far as well as show a diff but I can only think of one benefit of relying on **ALL** man pages being under man/ and that's convenience. But will the average user need to see the man pages for the dbg or jparse or dyn_array? These will eventually be their own repo (and one already is). Now if we do this it will make it harder to have the dbg/Makefile in this repo the same as the other one because it requires rules in man/Makefile like:

```makefile
#########################################################
# rules that invoke Makefile rules in other directories #
#########################################################

../dbg/dbg.3:
        @${MAKE} -C ../dbg man/man3/dbg.3

../dyn_array/dyn_array.3:
        @${MAKE} -C ../dyn_array man/man3/dyn_array.3

../jparse/man/jparse.1:
        @${MAKE} -C ../jparse man/man1/jparse.1

../jparse/man/jparse_test.8:
        @${MAKE} -C ../jparse man/man8/jparse_test.8

```

so that those Makefiles can have the same rules which might or might not be needed normally (I don't think they would be?). If we just had the man/Makefile install rule do a make install in the other directories it would dramatically simplify this.

What do you think?

I did make it so mkiocccentry/dbg/Makefile will install all man pages with `make install` in a commit I made but I think before I continue doing what I have in place (working changes) I will hold off. For now I don't think I need to include a diff or attach the file but if you think it will be of use I can do that.

**Comment by lcn2** [comment]
The `man/` sub-directory, after `make all`, holds all of the man pages, both those that **ONLY** apply to the _mkiocccentry repo_ AND those pages  copied from `dbg/man/`, `dyn_array/man/`, and `jparse/man/`.  The latter copying of man page is for convenience of the user, only.

We think that `make install` under `dbg/`, `dyn_array/`, and `jparse/` should also install their own man pages via their own Makefiles.

The question we see is, should this also happen for `soup/' and `test_ioccc/` as well?  I.e., should the man pages associated with their sub-directories live under `soup/man/` and `test_ioccc/man/`**AND** their respective Makefiles also install their man pages?

If the answer to the above question is yes, then perhaps the _chkentry(1)_, _iocccsize(1)_, _bug_report(1)_, _txzchk(1)_ could live under `soup/man/` (or perhaps at the top level directory).  And if this were to happen, then there would be no more "original" man pages left under `man/`.  If this is to be the case, then we propose that we remove the `man/` directory as a **PERMANENT** part of the _mkiocccentry repo_.

If the answer to the above question is yes, and we wanted to maintain the convince of a place to hold all of the man pages, then the top level `Makefile` could have a simple rule that creates a `man/` directory and copies all other man pages into it.  Now the Makefiles of the sub-directories would still install man pages as part of their `make install` rule.  And `.gitignore` would exclude `man` itself from the repo.

**Comment by lcn2** [comment]
fixed in commit b4688b76c05cdbf81eb7999a1336a7468762823a

**Comment by xexyl** [comment]
I'm sorry. I totally missed this comment. I'm not sure what was changed in the commit you made so I'll just comment here.

> The `man/` sub-directory, after `make all`, holds all of the man pages, both those that **ONLY** apply to the _mkiocccentry repo_ AND those pages copied from `dbg/man/`, `dyn_array/man/`, and `jparse/man/`. The latter copying of man page is for convenience of the user, only.
> 

The problem I have is that this does not seem to be done. After a `make clobber all` the `man/` is missing. Now we know where the man pages are but most people would think to check man/, right?

> We think that `make install` under `dbg/`, `dyn_array/`, and `jparse/` should also install their own man pages via their own Makefiles.
> 

I think that those directors should have their own man pages yes. This is especially useful since at least one of them is a repo outside here and the others will become one. Having these not require the master Makefile is definitely necessary for that reason.

> The question we see is, should this also happen for `soup/' and `test_ioccc/`as well? I.e., should the man pages associated with their sub-directories live under`soup/man/`and`test_ioccc/man/`**AND** their respective Makefiles also install their man pages?
> 

I'm honestly not sure. I thought that having the majority of the man pages under `man/` already was a good idea. As long as the `make all` works to copy them to `man/` I don't see the harm though it seems that it might complicate the Makefiles. However as I noted above it does not copy the files to `man/`.


> If the answer to the above question is yes, then perhaps the _chkentry(1)_, _iocccsize(1)_, _bug_report(1)_, _txzchk(1)_ could live under `soup/man/` (or perhaps at the top level directory). And if this were to happen, then there would be no more "original" man pages left under `man/`. If this is to be the case, then we propose that we remove the `man/` directory as a **PERMANENT** part of the _mkiocccentry repo_.

I think it makes more sense to have them under `mkiocccentry/man` - especially since the other man pages are in the subdirectories of their respective tools. After all `txzchk` is. under `mkiocccentry/` itself so it makes more sense that it would be under `mkiocccentry/man/man1`.

> 
> If the answer to the above question is yes, and we wanted to maintain the convince of a place to hold all of the man pages, then the top level `Makefile` could have a simple rule that creates a `man/` directory and copies all other man pages into it. Now the Makefiles of the sub-directories would still install man pages as part of their `make install` rule. And `.gitignore` would exclude `man` itself from the repo.

I do think `mkiocccentry/man` should exist at least after `make all`. Now I can see the reason of not having the `jparse/` and `dbg` and `dyn_array` man pages in `mkiocccentry/man` (for the reason I mentioned above) but I'm less sure of this for the main tools. I think having `soup/Makefile` and any other Makefile being required to do extra work just complicates things unnecessarily. But what really matters I think is that it works and after a compilation a `mkiocccentry/man` directory exists with _all_ the man pages even those of the separate directories.

There's certainly an argument for not originally having them in `mkiocccentry/man`: just like when you compile software in the old way (which I both have fond memories of and also don't exactly miss - same with ld preloading) it would sometimes (for some packages) build the man pages and then `make install` would copy them etc. But what I think is important in any event is that after `make all` `mkiocccentry/man` exists which it unfortunately doesn't.



> 
> ### UPDATE 0
> So the proposal would be:
> 
> * the install rule of each sub-directory to install the man pages under that sub-directory.
> * man pages that current live under `man/` would be moved elsewhere
> * the `man/` directory would be removed from the repo
> * "man" would be added to the `.gitignore` file
> 
> Optionally:
> 
> * the top level Makefile could have a rule that forms a "user convenience" `man/` tree by copying man pages from other sub-directories

Whether this option is the best way or not I don't know but what I think is important is that it's done somehow.

**Comment by xexyl** [comment]
I've discovered the problem but I don't know what you want to do about it so I'll make a new issue now. Then I'm done for the day most likely.

**Comment by xexyl** [comment]
> I've discovered the problem but I don't know what you want to do about it so I'll make a new issue now. Then I'm done for the day most likely.

I fixed this part and opened a new issue explaining it. Issue at https://github.com/ioccc-src/mkiocccentry/issues/471#issue-1524592580 and commit that fixes it is 9f3b06729364d9c9dec860584a6fc9338098e8a8.


The funny thing is that running `make bug_report` would work right but doing `make clobber all` would remove it but not recreate it!
